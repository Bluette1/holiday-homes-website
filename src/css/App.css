.image-area {
  height: 560px;
}

.imgContainer :hover {
  border: solid 4px #ecbc1c;
}

.imgContainer :focus {
  border: solid 4px #d8f167;
}

button,
input {
  overflow: visible;
  border-radius: 20px;
  outline: none;
  border: #00f;
  text-align: center;
  vertical-align: middle;
  padding: 0.375rem 0.75rem;
}

.btn {
  border-radius: 20px;
}

@media (max-width: 576px) {
  .image-area {
    height: 330px;
  }
}

@media screen and (max-width: 390px) {
  .signin-mssge {
    font-size: 0.65rem;
  }
}
