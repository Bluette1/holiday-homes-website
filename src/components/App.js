import React, { useState } from 'react';
import { Redirect } from 'react-router';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import HolidayHomesList from '../containers/HolidayHomesList';
import FavouritesList from '../containers/FavouritesList';
import SearchResultsList from '../containers/SearchResultsList';
import HolidayHomeForm from '../containers/HolidayHomeForm';
import Details from './HolidayHomeDetails';
import Navbar from './Navbar';
import User from './User';
import 'bootstrap/dist/css/bootstrap.min.css';
import '../css/App.css';

const App = ({ user }) => {
  const [renderFavourites, setRenderFavourites] = useState(false);
  const [renderDetails, setRenderDetails] = useState(false);
  const [holidayHome, setHolidayHome] = useState(null);
  const [favouriteId, setFavouriteId] = useState(null);
  const [renderNewForm, setRenderNewForm] = useState(false);
  const [renderUser, setRenderUser] = useState(false);
  const [renderSearchResults, setRenderResults] = useState({ show: false, params: '' });

  const showFavourites = (value = true) => {
    setRenderFavourites(value);
  };

  const showDetails = (value = true, holidayHome, favouriteId) => {
    setHolidayHome(holidayHome);
    setFavouriteId(favouriteId);
    setRenderFavourites(false);
    setRenderUser(false);
    setRenderDetails(value);
  };

  const showNewHolidayHome = (value = true) => {
    setRenderNewForm(value);
  };

  const showSearchResults = (show = true, params = '') => {
    setRenderResults({ show, params });
  };

  const showUser = (value = true) => {
    setRenderUser(value);
  };

  if (!user) {
    return <Redirect to="/login" />;
  }
  if (renderFavourites) {
    return (
      <div>
        <Navbar
          showFavourites={showFavourites}
          showNewHolidayHome={showNewHolidayHome}
          showDetails={showDetails}
          showUser={showUser}
          showSearchResults={showSearchResults}
        />
        <FavouritesList showFavourites={showFavourites} showDetails={showDetails} />
      </div>
    );
  }

  if (renderDetails) {
    return (
      <div>
        <Navbar
          showFavourites={showFavourites}
          showNewHolidayHome={showNewHolidayHome}
          showDetails={showDetails}
          showUser={showUser}
          showSearchResults={showSearchResults}
        />
        <Details holidayHomeObj={holidayHome} favouriteId={favouriteId} showDetails={showDetails} />
      </div>
    );
  }

  if (renderNewForm) {
    return (
      <div>
        <Navbar
          showFavourites={showFavourites}
          showNewHolidayHome={showNewHolidayHome}
          showDetails={showDetails}
          showUser={showUser}
          showSearchResults={showSearchResults}
        />
        <HolidayHomeForm showFavourites={showFavourites} showNewHolidayHome={showNewHolidayHome} />
      </div>
    );
  }

  if (renderUser) {
    return (
      <div>
        <Navbar
          showFavourites={showFavourites}
          showNewHolidayHome={showNewHolidayHome}
          showDetails={showDetails}
          showUser={showUser}
          showSearchResults={showSearchResults}
        />
        <User user={user} showUser={showUser} />
      </div>
    );
  }

  if (renderSearchResults.show) {
    return (
      <div>
        <Navbar
          showFavourites={showFavourites}
          showNewHolidayHome={showNewHolidayHome}
          showDetails={showDetails}
          showUser={showUser}
          showSearchResults={showSearchResults}
          showForm={false}
        />
        <SearchResultsList
          params={renderSearchResults.params}
          showSearchResults={showSearchResults}
          showDetails={showDetails}
        />
      </div>
    );
  }

  return (
    <div>
      <Navbar
        showFavourites={showFavourites}
        showNewHolidayHome={showNewHolidayHome}
        showDetails={showDetails}
        showUser={showUser}
        showSearchResults={showSearchResults}
      />
      <h3 className="mb-4 mt-2 text-center">Holiay homes</h3>
      <HolidayHomesList showDetails={showDetails} />
    </div>
  );
};
App.propTypes = {
  user: PropTypes.objectOf(PropTypes.any),
};

App.defaultProps = {
  user: {},
};

export default connect(state => ({ user: state.user }))(App);
