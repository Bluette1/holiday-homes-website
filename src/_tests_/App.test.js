import React from 'react';
import {
  render, screen, cleanup, waitFor,
} from '@testing-library/react';
import { Provider } from 'react-redux';
import { ConnectedRouter } from 'connected-react-router';
import axios from 'axios';
import App from '../components/App';
import configureStore, { history } from '../store';
import { httpProtocol, host, port } from '../envVariables';

jest.mock('axios');

afterEach(cleanup);

const store = configureStore({
  user: { username: 'MaryS', authentication_token: 'token' },
});

const AppWithStore = () => (
  <Provider store={store}>
    <React.StrictMode>
      <ConnectedRouter history={history}><App /></ConnectedRouter>
    </React.StrictMode>
  </Provider>
);
test('renders the app', async () => {
  axios.get.mockImplementation(url => {
    switch (url) {
      case `${httpProtocol}://${host}:${port}/categories`:
        return Promise.resolve({ data: { categories: [] } });
      default:
        return Promise.resolve({ data: [] });
    }
  });
  render(<AppWithStore />);
  await waitFor(() => {
    const holidayHomesTitle = screen.getByText(/Holiday-Homes/i);
    expect(holidayHomesTitle).toBeInTheDocument();
  });
});
