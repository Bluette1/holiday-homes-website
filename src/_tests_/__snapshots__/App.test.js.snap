// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the app 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="d-flex justify-content-between"
        >
          <nav
            class="navbar navbar-expand-lg navbar-light"
          >
            <span
              class="d-lg-sm d-none navbar-brand"
            >
              Holiday-Homes
            </span>
            <button
              aria-controls="basic-navbar-nav"
              aria-label="Toggle navigation"
              class="navbar-toggler collapsed"
              type="button"
            >
              <span
                class="navbar-toggler-icon"
              />
            </button>
            <div
              class="navbar-collapse collapse"
              id="basic-navbar-nav"
            >
              <div
                class="mr-auto navbar-nav"
              >
                <a
                  class="nav-link"
                  href="#"
                  role="button"
                >
                  Home
                </a>
                <a
                  class="nav-link"
                  href="#"
                  role="button"
                >
                  <select
                    class="category-filter"
                    data-testid="category-filter"
                  >
                    <option>
                      CATEGORIES
                    </option>
                    <option
                      value="All"
                    >
                      All
                    </option>
                  </select>
                </a>
                <div
                  class="dropdown nav-item"
                >
                  <a
                    aria-expanded="false"
                    aria-haspopup="true"
                    class="dropdown-toggle nav-link"
                    href="#"
                    id="basic-nav-dropdown"
                    role="button"
                  >
                    MaryS
                  </a>
                </div>
              </div>
            </div>
          </nav>
          <nav>
            <button
              aria-controls="collapseExample"
              aria-expanded="false"
              class="m-2 p-2 pl-3 pr-3 search-btn"
              data-target="#collapseTarget"
              data-toggle="collapse"
              type="button"
            >
              <span
                class=""
              >
                <i
                  aria-hidden="true"
                  class="fa fa-search"
                />
              </span>
            </button>
            <div
              class="collapse m-5 p-5"
              id="collapseTarget"
            >
              <form>
                <input
                  class="mr-sm-2 search-input"
                  placeholder="Search by title"
                  type="text"
                />
                <button
                  aria-labelledby="button-label"
                  class="btn btn-primary mt-2"
                  type="submit"
                >
                  Search
                </button>
              </form>
            </div>
          </nav>
        </div>
        <h3
          class="mb-4 mt-2 text-center"
        >
          Holiday homes
        </h3>
        <p
          class="no-holiday-homes"
        >
          No holiday homes were found
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="d-flex justify-content-between"
      >
        <nav
          class="navbar navbar-expand-lg navbar-light"
        >
          <span
            class="d-lg-sm d-none navbar-brand"
          >
            Holiday-Homes
          </span>
          <button
            aria-controls="basic-navbar-nav"
            aria-label="Toggle navigation"
            class="navbar-toggler collapsed"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="navbar-collapse collapse"
            id="basic-navbar-nav"
          >
            <div
              class="mr-auto navbar-nav"
            >
              <a
                class="nav-link"
                href="#"
                role="button"
              >
                Home
              </a>
              <a
                class="nav-link"
                href="#"
                role="button"
              >
                <select
                  class="category-filter"
                  data-testid="category-filter"
                >
                  <option>
                    CATEGORIES
                  </option>
                  <option
                    value="All"
                  >
                    All
                  </option>
                </select>
              </a>
              <div
                class="dropdown nav-item"
              >
                <a
                  aria-expanded="false"
                  aria-haspopup="true"
                  class="dropdown-toggle nav-link"
                  href="#"
                  id="basic-nav-dropdown"
                  role="button"
                >
                  MaryS
                </a>
              </div>
            </div>
          </div>
        </nav>
        <nav>
          <button
            aria-controls="collapseExample"
            aria-expanded="false"
            class="m-2 p-2 pl-3 pr-3 search-btn"
            data-target="#collapseTarget"
            data-toggle="collapse"
            type="button"
          >
            <span
              class=""
            >
              <i
                aria-hidden="true"
                class="fa fa-search"
              />
            </span>
          </button>
          <div
            class="collapse m-5 p-5"
            id="collapseTarget"
          >
            <form>
              <input
                class="mr-sm-2 search-input"
                placeholder="Search by title"
                type="text"
              />
              <button
                aria-labelledby="button-label"
                class="btn btn-primary mt-2"
                type="submit"
              >
                Search
              </button>
            </form>
          </div>
        </nav>
      </div>
      <h3
        class="mb-4 mt-2 text-center"
      >
        Holiday homes
      </h3>
      <p
        class="no-holiday-homes"
      >
        No holiday homes were found
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
